### PR TITLE
fix: Short circuit on exipiration

### PR DESCRIPTION
**Pull request by PirosB3**
When a RFQ order is expired, mark it directly as expired

### CODE REVIEW

File: src/services/rfqm_service.ts [hunk 1]
```diff
@@ -244,7 +244,7 @@ export class RfqmService {
     }
 
     // Returns a failure status for invalide jobs and null if the job is valid.
```
Comment (rhinodavid) [inline src/services/rfqm_service.ts:246]:
```suggestion
    // Returns a failure status for invalid jobs and null if the job is valid.
```

File: src/services/rfqm_service.ts [hunk 2]
```diff
@@ -262,6 +262,19 @@ export class RfqmService {
             return RfqmJobStatus.FailedValidationNoFee;
         }
 
+        if (order.type === RfqmOrderTypes.V4Rfq) {
+
+            // Orders can expire if any of the following happen:
+            // 1) workers are backed up
+            // 2) an RFQM order broke during submission and the order is stuck in the queue for a long time.
+            const v4Order = order.order;
+            const expiryInMs = new BigNumber(v4Order.expiry).times(ONE_SECOND_MS);
```
Comment (rhinodavid) [inline src/services/rfqm_service.ts:271]: maybe `expiryTimeMs`

File: src/services/rfqm_service.ts [hunk 3]
```diff
@@ -262,6 +262,19 @@ export class RfqmService {
...
+            console.log()
```
Comment (rhinodavid) [inline src/services/rfqm_service.ts:272]: oh no

Comment (PirosB3) [reply src/services/rfqm_service.ts:272]: ugh!! forgot the log msg! putting it in now

File: test/services/rfqm_service_test.ts [hunk 4]
```diff
@@ -91,6 +91,77 @@ const buildRfqmServiceForUnitTest = (
 };
 
 describe('RfqmService', () => {
+
+    it('should be able to validate jobs and mark them as expired', () => {
```
Comment (rhinodavid) [inline test/services/rfqm_service_test.ts:95]:
```suggestion
    it('should validate jobs and mark them as expired', () => {
```

File: test/services/rfqm_service_test.ts [hunk 5]
```diff
@@ -91,6 +91,77 @@ const buildRfqmServiceForUnitTest = (
...
+            updatedAt: new Date(),
+            workerAddress: '',
+            orderHash: '',
+            metaTransactionHash: '',
+            calldata: '0x000',
+            makerUri: 'http://foo.bar',
+        };
+
+        // First test, order not expired
+        let response = RfqmService.validateJob({
```
Comment (rhinodavid) [inline test/services/rfqm_service_test.ts:130]: didn't you make this private

**Review by rhinodavid — APPROVED** [review]

**Pull request merged**